fuzz/fuzz_targets/compare_xz.rs

```diff
@@ -0,0 +1,36 @@
+#![no_main]
+#[macro_use]
+extern crate libfuzzer_sys;
+extern crate lzma_rs;
+extern crate xz2;
+
+use std::io::Read;
+use lzma_rs::error::Result;
+use xz2::stream;
+
+fn decode_xz_lzmars(compressed: &[u8]) -> Result<Vec<u8>> {
+    let mut bf = std::io::Cursor::new(compressed);
+    let mut decomp: Vec<u8> = Vec::new();
+    lzma_rs::xz_decompress(&mut bf, &mut decomp)?;
+    Ok(decomp)
+}
+
+fn decode_xz_xz2(compressed: &[u8]) -> Result<Vec<u8>> {
+    let bf = std::io::Cursor::new(compressed);
+    let mut decomp: Vec<u8> = Vec::new();
+    // create new XZ decompression stream with 8Gb memory limit and checksum verification disabled
+    let xz_stream = stream::Stream::new_stream_decoder(8589935000, stream::IGNORE_CHECK).expect("Failed to create stream");
```

Comment:
Where does the exact value `8589935000` come from? Is it better represented in hexadecimal form? Is it a magic value provided by the `xz2` crate, or by liblzma?

Reply:
That's 8 gibibytes in bytes. Frankly the value is quite arbitrary, I just wanted to have something as a starting point.

Reply:
I've rewritten it in a more readable way

Reply:
Well, `8*1024*1024*1024` is `8589934592`, not `8589935000` so that surprised me at first. I think `0x2_0000_0000` would be more readable (`0x2` followed by 8 nibbles of zeros = 32 bits of zeros).